File: android/lib/features/post/post_metadata_header.dart
import 'package:flutter/material.dart';

import '../../core/utils.dart';
import '../../models/post.dart';
import '../../theme/app_config.dart';

class PostMetadataHeader extends StatelessWidget {
  final PostDetail post;

  const PostMetadataHeader({super.key, required this.post});

  static const _months = [
    'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun',
    'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec',
  ];

  String _formatDate(DateTime? dt) {
    if (dt == null) return '';
    return '${_months[dt.month - 1]} ${dt.day}, ${dt.year}';
  }

  @override
  Widget build(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        // Source chip + quality badge row
        Row(
          children: [
            Container(
              padding: const EdgeInsets.symmetric(
                horizontal: AppConfig.spacingSm,
                vertical: AppConfig.spacingXs,
              ),
              decoration: BoxDecoration(
                color: AppConfig.primaryMuted,
                borderRadius: BorderRadius.circular(AppConfig.radiusFull),
              ),
              child: Text(
                post.sourceName,
                style: const TextStyle(
                  color: AppConfig.primary,
                  fontSize: 12,
                  fontWeight: FontWeight.w500,
                ),
              ),
            ),
            const SizedBox(width: AppConfig.spacingSm),
            // Quality badge
            Container(
              padding: const EdgeInsets.symmetric(
                horizontal: AppConfig.spacingSm,
                vertical: AppConfig.spacingXs,
              ),
              decoration: BoxDecoration(
                color: AppConfig.qualityColor(post.qualityScore)
                    .withValues(alpha: 0.2),
                borderRadius: BorderRadius.circular(AppConfig.radiusFull),
              ),
              child: Text(
                AppConfig.qualityGrade(post.qualityScore),
                style: TextStyle(
                  color: AppConfig.qualityColor(post.qualityScore),
                  fontSize: 12,
                  fontWeight: FontWeight.bold,
                ),
              ),
            ),
          ],
        ),
        const SizedBox(height: AppConfig.spacingMd),
        // Metadata line
        Wrap(
          spacing: AppConfig.spacingMd,
          runSpacing: AppConfig.spacingXs,
          children: [
            if (post.author != null)
              _metaItem(Icons.person_outline, post.author!),
            if (post.publishedAt != null)
              _metaItem(Icons.calendar_today, _formatDate(post.publishedAt)),
            if (post.wordCount != null)
              _metaItem(Icons.article_outlined, '${post.wordCount} words'),
            if (post.audioDurationSecs != null && post.audioStatus == 'ready')
              _metaItem(Icons.headphones,
                  formatDurationFromSeconds(post.audioDurationSecs)),
          ],
        ),
        // Tags
        if (post.tags.isNotEmpty) ...[
          const SizedBox(height: AppConfig.spacingMd),
          Wrap(
            spacing: AppConfig.spacingSm,
            runSpacing: AppConfig.spacingSm,
            children: post.tags
                .map((tag) => Chip(
                      label: Text(tag),
                      materialTapTargetSize: MaterialTapTargetSize.shrinkWrap,
                      visualDensity: VisualDensity.compact,
                    ))
                .toList(),
          ),
        ],
      ],
    );
  }

  Widget _metaItem(IconData icon, String text) {
    return Row(
      mainAxisSize: MainAxisSize.min,
      children: [
        Icon(icon, size: 14, color: AppConfig.mutedText),
        const SizedBox(width: AppConfig.spacingXs),
        Text(
          text,
          style: const TextStyle(color: AppConfig.mutedText, fontSize: 13),
        ),
      ],
    );
  }
}

File: ios/TechBlogCatchup/Features/PostDetail/PostMetadataHeader.swift
import SwiftUI

struct PostMetadataHeader: View {
    let post: PostDetail

    @Environment(\.theme) private var theme

    var body: some View {
        VStack(alignment: .leading, spacing: theme.spacingSm) {
            // Title
            Text(post.title)
                .font(theme.titleLargeFont)
                .foregroundColor(theme.textPrimary)
                .fixedSize(horizontal: false, vertical: true)

            // Source badge + author
            HStack(spacing: theme.spacingSm) {
                ThemedBadge(text: post.sourceName, color: theme.primary)

                if let author = post.author, !author.isEmpty {
                    Text("by \(author)")
                        .font(theme.captionFont)
                        .foregroundColor(theme.textSecondary)
                }
            }

            // Metadata row
            HStack(spacing: theme.spacingMd) {
                if let publishedAt = post.publishedAt {
                    metadataItem(icon: "calendar", text: publishedAt.formatTimeAgo())
                }

                if let wordCount = post.wordCount {
                    metadataItem(icon: "doc.text", text: "\(wordCount) words")
                }

                if let secs = post.audioDurationSecs {
                    metadataItem(icon: "clock", text: Date.formatDuration(seconds: secs))
                }
            }

            // Quality badge
            if let score = post.qualityScore {
                HStack(spacing: theme.spacingSm) {
                    Text("Quality:")
                        .font(theme.captionFont)
                        .foregroundColor(theme.textTertiary)

                    ThemedBadge(
                        text: qualityGrade(score: score),
                        color: qualityColor(score: score)
                    )
                }
            }

            // Tags
            if !post.tags.isEmpty {
                ScrollView(.horizontal, showsIndicators: false) {
                    HStack(spacing: theme.spacingXs) {
                        ForEach(post.tags, id: \.self) { tag in
                            TagBadgeView(tag: tag)
                        }
                    }
                }
            }
        }
    }

    private func metadataItem(icon: String, text: String) -> some View {
        HStack(spacing: 4) {
            Image(systemName: icon)
                .font(.system(size: 12))
            Text(text)
                .font(theme.captionFont)
        }
        .foregroundColor(theme.textTertiary)
    }

    private func qualityGrade(score: Int) -> String {
        switch score {
        case 80...100: return "A"
        case 60..<80: return "B"
        case 40..<60: return "C"
        case 20..<40: return "D"
        default: return "F"
        }
    }

    private func qualityColor(score: Int) -> Color {
        switch score {
        case 80...100: return theme.success
        case 60..<80: return theme.successLight
        case 40..<60: return theme.warning
        default: return theme.error
        }
    }
}
